Modules/Common/AvalancheLikeCrossChainModule.php

```diff
@@ -0,0 +1,308 @@
+
+    final public function post_post_initialize()
+    {
+        $this->unpacker_handle = envm(
+            $this->module,
+            'UNPACKER_SERVICE',
+            new DeveloperError('Unpacker service is not set in the config')
+        );
+
+        $this->asset_info_handle = envm(
+            $this->module,
+            'ASSETS_SERVICE',
+            new DeveloperError('Assets service is not set in the config')
+        );
+
+        if (is_null($this->main_token_descr))
+            throw new DeveloperError('`main_token_descr` is not set (developer error)');
+
+        if (is_null($this->sister_networks))
+            throw new DeveloperError('`sister_networks` is not set (developer error)');
+    }
+
+    final public function pre_process_block($block_id)
+    {
+        $multi_curl = [];
+
+        $multi_curl[] = requester_multi_prepare($this->select_node(),
+            params: ['method'  => 'eth_getBlockByNumber',
+                     'params'  => [to_0xhex_from_int64($block_id), true],
+                     'id'      => 0,
+                     'jsonrpc' => '2.0',
+            ], timeout: $this->timeout);
+
+        $curl_results = requester_multi($multi_curl, limit: envm($this->module, 'REQUESTER_THREADS'), timeout: $this->timeout);
+        $r1 = requester_multi_process($curl_results[0], result_in: 'result');
+
+        $block_time = $r1['timestamp'];
+
+        $atomic_transactions = [];
+        if (array_key_exists('blockExtraData', $r1)) {
+            if (str_starts_with($r1['blockExtraData'], '0x') && strlen($r1['blockExtraData']) > 2) {
+                $atomic_transactions = $this->middleware_unpack_atomics($block_time, $r1['blockExtraData']);
+            }
+        }
+
+        [$events, $currencies_used] = $this->prepare_atomic_events($atomic_transactions);
+
+        $this->process_events($block_id, $events, $block_time);
+
+        $currencies_used = check_existing_currencies($currencies_used, $this->currency_format);
+
+        $this->process_currencies($currencies_used);
+    }
+
+    private function middleware_unpack_atomics($timestamp, $blob)
+    {
+        return requester_single($this->unpacker_handle, params: [
+            'method' => 'getAtomics',
+            'params' => [
+                'timestamp'      => $timestamp,
+                'blockExtraData' => $blob
+            ],
+            'id' => 0,
+            'jsonrpc' => '2.0',
+        ],
+        result_in: 'result', timeout: $this->timeout);
+    }
+
+    private function parse_transaction_legs(
+        &$tx_legs,
+        $hash,
+        $sort_outer,
+        $sort_inner,
+        $signum,
+        $extra,
+        &$atomic_events,
+        &$currencies_used
+    )
+    {
+        foreach ($tx_legs as $tx_leg)
+        {
+            $atomic_events[] = [
+                'transaction'         => $hash,
+                'address'             => (is_null($extra)) ? $tx_leg['addr'] : ($this->sister_networks[$tx_leg['blockchain']] ?? '0x00'),
```

Comment:
what is the case for `0x00`?
The address can only be a C-chain address or the specific addresses X-chain, P-Chain, can't it?

Reply:
This is a dummy value. Should avax add support for more crosschain transfers besides C, X and P - module won't crash, it would be easy to find such entry in the database and update the module accordingly